### PR TITLE
Drop Python 3.9 support

### PR DESCRIPTION
**Pull request by seapagan**
This PR drops Python 3.9 support and moves the project floor to Python 3.10.

It updates package metadata, Ruff and mypy targets, the GitHub Actions test matrix, docs, and generated dependency exports. It also updates the dev tooling stack, including pytest 9 and mypy 2, and applies the Python 3.10 typing cleanups required by the newer Ruff rules.

While validating the dependency updates, pytest 9 surfaced SQLite ResourceWarning noise from tests that left connections to be finalized by GC. The test fixtures and related edge cases now close database handles deterministically, with a small finalizer backstop for abandoned sync database instances.

## Summary by CodeRabbit

* **New Features**
  * Introduced async functionality (use cautiously in early releases).

* **Documentation**
  * Minimum Python requirement raised to 3.10+; note that 3.9 support ends after 0.20.0.
  * Clarified nullable field detection and foreign-key behaviour; added async-related caution.

* **Chores**
  * CI updated (removed Python 3.9, added Python 3.14).
  * Development dependencies and test infrastructure refreshed; tests now ensure DB connections are closed and finalisation cleans up unclosed connections.

[Review Change Stack](https://app.coderabbit.ai/change-stack/seapagan/sqliter-py/pull/150)

### CODE REVIEW

File: tests/test_complex_types.py [hunk 1]
```diff
@@ -193,9 +193,7 @@ def test_type_validation(
         self,
         model_instance: ComplexTypesModel,
         field_name: str,
-        invalid_value: Union[
-            list[Any], dict[Any, Any], set[Any], tuple[Any, ...]
-        ],
+        invalid_value: list[Any] | dict[Any, Any] | set[Any] | tuple[Any, ...],
```
[the remark this answers is not in the document]
Comment (seapagan) [reply tests/test_complex_types.py]: actually, easier is to replace the entire type with `object`

**Comment by seapagan** [comment]
@coderabbitai review
